src/xviewer-image.c

```diff
@@ -760,8 +757,18 @@ xviewer_image_get_file_info (XviewerImage *img,
                 }
             }
 
+            mime_type_by_extn = g_strdup (g_file_info_get_content_type (file_info));
+
             if (file_type_unknown)
-                *mime_type = g_strdup (g_file_info_get_content_type (file_info));
+               *mime_type = mime_type_by_extn;
+            else
+            {
+                if ((strstr(*mime_type,"application/") == *mime_type) &&
+                    (strstr(mime_type_by_extn,"image/") == mime_type_by_extn))
+                    *mime_type = g_strdup(mime_type_by_extn);
+            }
+
+            g_free(mime_type_by_extn);
         }
```

Comment:
Just a nitpick - please put a space between the function names and the open parentheses: `strstr (*mime_type..`

Reply:
I'll do this then upload the edited file.

Reply:
Is there a problem with this PR? I updated the file to put a space between the function names and the open parentheses and sorted the g_strdup (mime_type_by_extn) error and then uploaded the modified file.

The question of mime_type not being initialised was answered above (it's a parameter to the function and therefore must not be initialised within the function).

If anything else needs changing just let me know.

Reply:
Sorry nothing wrong, I've just been neglecting to come back to this when reviewing prs.

Reply:
Apologies - I wasn't meaning to hassle you but I wondered if we were in the situation where you were waiting for me to upload the file and I was waiting for you to merge the PR and so nothing happened   :-)